packages/flutter/lib/src/material/selectable_text.dart

```diff
@@ -696,6 +702,7 @@ class _SelectableTextState extends State<SelectableText> with AutomaticKeepAlive
     );
 
     return Semantics(
+      label: widget.semanticsLabel,
       onLongPress: () {
```

Comment:
this need to excludesemantics if semantics label is present

Reply:
Adding `excludeSemantics: widget.semanticsLabel != null` results in 
```
expect(semantics, hasSemantics(TestSemantics.root(
      children: <TestSemantics>[
        TestSemantics.rootChild(
          id: 1,
          textDirection: TextDirection.ltr,
          label: 'German greeting for good day',
          value: 'Guten Tag',
          actions: <SemanticsAction>[
            SemanticsAction.longPress,
          ],
          flags: <SemanticsFlag>[
            SemanticsFlag.isTextField,
            SemanticsFlag.isReadOnly,
            SemanticsFlag.isMultiline,
          ],
        )
      ],
    ), ignoreTransform: true, ignoreRect: true));
```
becoming

```
    expect(semantics, hasSemantics(TestSemantics.root(
      children: <TestSemantics>[
        TestSemantics.root(
          children: <TestSemantics>[
            TestSemantics(
              id: 1,
              actions: <SemanticsAction>[SemanticsAction.longPress],
              label: 'German greeting for good day',
              textDirection: TextDirection.ltr,
            ),
          ],
        )
      ],
    ), ignoreTransform: true, ignoreRect: true));
  });
```

which doesn't seem quite right. I'm not super familiar with `Semantics`, any ideas?

Reply:
The former will make screen reader announce `German greeting for good day, Guten Tag` the latter will announce
`German greeting for good day`, I believe the later is the correct behavior?

I also took at the Text widget, it looks like the `semanticsLabel` will also overwrite the text in semantics

Reply:
Indeed, yes. I thought the missing flags were a problem, but they're just not included in the suggested "would have matched" output.

Reply:
I dont think this comment is resolved, why is the pr merged?

The semanticsLabel will makes append to the begining of the selectable text when announce by screen reader instead of replacing it.

Reply:
Woops, forgot a commit. Fixing forward with https://github.com/flutter/flutter/pull/90136